bonus_model.py

```diff
@@ -1,15 +1,32 @@
 """Define your architecture here."""
 import torch
-from models import SimpleNet
-
+from torch import nn
+import torchvision.models as torch_models
 
 def my_bonus_model():
     """Override the model initialization here.
 
     Do not change the model load line.
     """
     # initialize your model:
-    model = SimpleNet()
+    model = ResnetBasedModel()
     # load your model using exactly this line (don't change it):
     model.load_state_dict(torch.load('checkpoints/bonus_model.pt')['model'])
     return model
+
+class ResnetBasedModel(nn.Module):
+    def __init__(self):
+        super().__init__()
+        self.encoder = torch_models.resnet18(pretrained=True)
+        self.encoder = nn.Sequential(*(list(self.encoder.children())[:-1]))
+
+        mlp = [
+            nn.Linear(512, 2),
+        ] 
+
+        self.fc = nn.Sequential(*mlp)
```

Comment:
couldn't we have used nn.Linear(512, 2) as-is?

Reply:
For sure, i was experimenting with different heads so it was more convenient to write it this way in case i wanted to add more layers, i still have a rogue comma there